railties/lib/rails/application.rb

```diff
@@ -219,7 +219,7 @@ def message_verifier(verifier_name)
     #     Rails.application.configure do
     #       config.middleware.use ExceptionNotifier, config_for(:exception_notification)
     #     end
-    def config_for(name, env: Rails.env)
+    def config_for(name, env: Rails.env, default: {})
```

Comment:
Should we add a test for the added default parameter?

Reply:
Yes

Reply:
When is this option useful? Seems strange to me to bypass the `config_for` file read entirely, just to get OrderedOptions returned. I've just had a case where I had a config file without an env split in it, I feel like that would be a better use case to solve.

Reply:
> When is this option useful? 

I made this an option because if like in #37800 the expected return value is an Array, it's weird to have a hash being returned as a default.

But it could also make sense to rethink what should be returned / raised on miss.

> I've just had a case where I had a config file without an env split in it, I feel like that would be a better use case to solve.

Not sure I understand what you mean here.

Reply:
I'd think `nil` would be the default return such that you could do `|| []` or `to_a` or `|| "…"` or `|| Rails.application.config.x.shared_secrets_value`. I don't like `default:` here because it's unclear from the calling code when it's applied. Does it mean ^ or are the shared keys added? If the latter you'd only get the shared keys but those aren't necessarily complete without extra configs on top.

> Not sure I understand what you mean here.

I ended up changing the code structure to formulate those defaults in code instead. So let's just disregard this for now.

Reply:
> I'd think `nil` would be the default return such that you could do

That's what I initially wonted to do, but I was afraid it would be considered a breaking change.